src/api/player.rs

```diff
@@ -80,3 +94,30 @@ pub fn get_player(
 
     Ok(Json(response))
 }
+
+#[put("/", format = "application/json", data = "<req>")]
+pub fn update_player(
+    req: Json<UpdatePlayerRequest>,
+    conn: DbConn,
+    _token: ApiToken,
+) -> Result<Json<PlayerResponse>, ApiError> {
+    let update_request = req.into_inner();
+
+    let old_player = Player::find(update_request.id, &conn)?;
+    let new_player = Player::update(
+        Player {
+            id: old_player.id,
+            alias: update_request.alias.unwrap_or(old_player.alias),
+            email: update_request.email.unwrap_or(old_player.email),
+            password: old_player.password,
+        },
+        &conn,
+    )?;
+
+    let response = PlayerResponse {
+        id: new_player.id,
+        alias: new_player.alias,
```

Comment:
email too?

Reply:
We don't return email in the player response for security/privacy reasons.